### PR TITLE
[3.2] Switch Embed.ly to oEmbed API

### LINKED ISSUE #6635: Replace baked-in embed.ly with something else
Recently, embed.ly changed their business model, so we can't use it in Bolt anymore. At least, not in the way we used to. We need to replace it asap. 

More information: [embedly faq](http://docs.embed.ly/docs/new-pricing-faq) / [blogpost with marketing bingo](https://blog.embed.ly/the-best-is-yet-to-come-f5aac2bd6b13)

Current options on the table are: 

 - 1. Fork over $99 per month to keep the service running as is. We're going to do this for May, to give us some time to implement another option.

 - 2. Switch over to (third party / hosted) noembed.com, urlembed.com or iframely.com. We can do this rather quickly, and include it in a 3.2.x release rather painlessly. 

 - 3. Provide configuration options in Bolt, so people can use either the "default" provider that is to be determined, or their own third-party or self-hosted option

 - 4. Integrate `embed/embed` with Bolt, so we can use an "internal" provider. 

 - 5. Self-host a noembed installation on embed.bolt.cm. This would seem "professional", and we're less dependant on third parties, but it will require extra maintenance from us on another module. Additionally, there might be privacy concerns, because people might question why embedded URL's are sent to a bolt.cm subdomain. 

Currently i'm leaning towards option 1 (for the short term), and get a 3.2.x out of the door with option 2 and 3. 
If we then also provide either option 4 or 5 later on, it can still work as a configurable option in Bolt: Use what we provide out-of-the-box, or select your own self-hosted or third-party solution.

### PR DESCRIPTION
**Pull request by GawainLynch**
Does the job, gets us through, relies on the SF Forms token to be valid in editcontent …
… RFC

Fixes #6635

### CODE REVIEW

File: app/src/js/widgets/fields/fieldVideo.js [hunk 1]
```diff
@@ -107,11 +107,13 @@
          */
         _update: function () {
             var self = this,
-                url = 'https://api.embed.ly/1/oembed',
+                url = bolt.conf('paths.async') + 'oembed/request',
```
Comment (CarsonF) [inline app/src/js/widgets/fields/fieldVideo.js]: I've been trying to get away from string concatenation with paths. Can you assign url generator result to a data attribute in twig and fetch that here?

File: src/Request/Oembed/GuzzleDispatcher.php [hunk 2]
```diff
@@ -0,0 +1,99 @@
+<?php
+
+namespace Bolt\Request\Oembed;
```
Comment (CarsonF) [inline src/Request/Oembed/GuzzleDispatcher.php]: I know this is making a request to (O)embed, but I don't think it belongs here. It has nothing to do with our Symfony HTTP Requests. Instead it's a service we are calling. I think `Bolt\(O)embed` would be better.

File: src/Controller/Async/Oembed.php [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+     * @param Url $url
+     *
+     * @return \Embed\Adapters\Adapter
+     */
+    protected function getUrlAdapter(Url $url)
+    {
+        $oEmbedFactory = $this->app['oembed.factory'];
+
+        return $oEmbedFactory($url);
+    }
```
Comment (CarsonF) [inline src/Controller/Async/Oembed.php]: I kinda think having our own service class that wraps the ombed api that the controller could call here would be better. Then the controller doesn't have to know anything about the `Embed\*` code. But if there's a rush don't worry about it.

Comment (GawainLynch) [inline src/Controller/Async/Oembed.php]: It needs to be a factory, and I have over spend hugely my allowed computer time today.

So if someone wants to pick it up tonight, go for it … but a) this is 3.2; b) urgent

File: src/Request/Oembed/GuzzleDispatcher.php (hunk 2 above)
Comment (GawainLynch) [inline src/Request/Oembed/GuzzleDispatcher.php]: Tomorrow, or anyone with push access

File: app/src/js/widgets/fields/fieldVideo.js (hunk 1 above)
Comment (GawainLynch) [inline app/src/js/widgets/fields/fieldVideo.js]: All of it is getting re-done … I just went with what is there … maybe tomorrow, but I think that will have to wait for 3.3 TBH

Comment (GawainLynch) [inline app/src/js/widgets/fields/fieldVideo.js]: Done :wink:

File: src/Controller/Async/Oembed.php (hunk 3 above)
Comment (GawainLynch) [inline src/Controller/Async/Oembed.php]: Done … I got a better sense of what you meant after sleeping on it, and yeah :smile:

File: src/Request/Oembed/GuzzleDispatcher.php (hunk 2 above)
Comment (GawainLynch) [inline src/Request/Oembed/GuzzleDispatcher.php]: Done

**Comment by bobdenotter** [comment]
Merging in! :shipit:

**Pull request merged**